codec/src/main/java/io/netty/handler/codec/LengthFieldBasedFrameDecoder.java

```diff
@@ -346,6 +346,52 @@ protected final void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object>
         }
     }
 
+    private void discardingTooLongFrame(ByteBuf in) {
+        long bytesToDiscard = this.bytesToDiscard;
+        int localBytesToDiscard = (int) Math.min(bytesToDiscard, in.readableBytes());
+        in.skipBytes(localBytesToDiscard);
+        bytesToDiscard -= localBytesToDiscard;
+        this.bytesToDiscard = bytesToDiscard;
+
+        failIfNecessary(false);
+    }
+
+    private void failOnNegativeLengthField(ByteBuf in, long frameLength) {
+        in.skipBytes(lengthFieldEndOffset);
+        throw new CorruptedFrameException(
+           "negative pre-adjustment length field: " + frameLength);
+    }
+
+    private void failOnFrameLengthLessThanLengthFieldEndOffset(ByteBuf in, long frameLength) {
+        in.skipBytes(lengthFieldEndOffset);
+        throw new CorruptedFrameException(
+           "Adjusted frame length (" + frameLength + ") is less " +
+              "than lengthFieldEndOffset: " + lengthFieldEndOffset);
+    }
+
+    private void exceededFrameLength(ByteBuf in, long frameLength) {
+        long discard = frameLength - in.readableBytes();
+        tooLongFrameLength = frameLength;
+
+        if (discard < 0) {
+            // buffer contains more bytes then the frameLength so we can discard all now
+            in.skipBytes((int) frameLength);
+        } else {
+            // Enter the discard mode and discard everything received so far.
+            discardingTooLongFrame = true;
+            bytesToDiscard = discard;
+            in.skipBytes(in.readableBytes());
+        }
+        failIfNecessary(true);
+    }
+
+    private void failOnFrameLengthLessThanInitialBytesToStrip(ByteBuf in, long frameLength, int frameLengthInt) {
```

Comment:
Can this be static?

Reply:
I wouldn't go out of the way to transform all methods to static, I was under the impression that it was a common coding style to use static helper / utility methods which are stateless.

Reply:
Got it...All the methods I've moved away from the decode are not stateless in their original form (copied from the original decode method), but can be transformed in that way, as I pointed in the last comment.